controllers/workspace/provision/deployment.go

```diff
@@ -41,13 +43,21 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
-var ContainerFailureStateReasons = []string{
+var containerFailureStateReasons = []string{
 	"CrashLoopBackOff",
 	"ImagePullBackOff",
 	"CreateContainerError",
 	"RunContainerError",
 }
 
+var unrecoverablePodEventReasons = []string{
+	"FailedMount",
```

Comment:
On some infrastructure, FailedMount or FailedScheduling can be an usual situation when you just need to be more patient and then on Che we suggest users to drop them from unrecoverable events list.
We should have it configurable in some way or let users annotate particular workspace, maybe get it from namespaced configmap settings if present.
WDYT?

Reply:
I feel like an annotation system might complicate things too much -- we should just do what we think is best (whether that's check or don't check).

One issue I've seen in Che before is that there can be a bit of a [thundering herd problem](https://en.wikipedia.org/wiki/Thundering_herd_problem) -- if I've got 10 workspaces waiting for the common PVC to be mountable, it can be very confusing because they'll randomly start up and have to be stopped until it gets to the one I want to actually use. 

One option that just occurred to me from looking at the Event struct a bit more -- what if we have a threshold for "number of times the event has occurred"? We could fail workspace start if e.g. we've seen the `FailedMount` even three times.

Reply:
Created https://github.com/devfile/devworkspace-operator/issues/550 to follow up